timely/src/worker.rs

```diff
@@ -16,11 +18,79 @@ use crate::progress::operate::Operate;
 use crate::dataflow::scopes::Child;
 use crate::logging::TimelyLogger;
 
+/// Progress mode. (???)
+#[derive(Debug, Clone, Copy, Eq, PartialEq)]
+pub enum ProgressMode {
+    /// ???
+    Eager,
+    /// ???
+    Demand,
+}
+
+impl Default for ProgressMode {
+    fn default() -> ProgressMode {
+        ProgressMode::Eager
+    }
+}
+
+impl FromStr for ProgressMode {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<ProgressMode, String> {
+        match s {
+            "eager" => Ok(ProgressMode::Eager),
+            "demand" => Ok(ProgressMode::Demand),
+            _ => Err(format!("unknown progress mode: {}", s)),
+        }
+    }
+}
+
+/// Worker configuration.
+#[derive(Debug, Default, Clone)]
+pub struct WorkerConfig {
+    /// The progress mode to use.
+    pub(crate) progress_mode: ProgressMode,
+    /// A map from parameter name to typed parameter values.
+    registry: HashMap<String, Arc<dyn Any + Send + Sync>>,
```

Comment:
Is there a non-obviously reason that this is an `Arc` instead of a `Box`?

Reply:
Alternately (if it is like, setting up the configs for everyone and I've missed that) is the `Send` required for `Arc`?

Reply:
Yeah, what we actually need is a clonable `Box<dyn Any>`, but I don't know how to make that happen easily types besides `Arc<dyn Any + Send + Sync>`, which is a bit overspecific. Specifically, I don't think `Sync` is important, but `Arc<T>` doesn't implement `Send` unless the underlying type implements `Sync` too.

I'm sure you could make something work with https://docs.rs/dyn-clone/1.0.4/dyn_clone/, but it's always a big pain in the butt in my experience.